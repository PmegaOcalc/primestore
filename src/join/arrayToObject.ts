export function arrayToObject<T extends any[]>(
  array: T
): Partial<Record<T[number], true>> {
  return array.reduce((obj, item) => {
    obj[item] = true;
    return obj;
  }, {});
}
