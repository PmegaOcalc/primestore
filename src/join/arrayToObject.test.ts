import { arrayToObject } from "./arrayToObject";
import { Statuses } from "./test.types";

describe("arrayToObject function", () => {
  test("to be defined", () => {
    expect(arrayToObject).toBeDefined();
  });

  test("renders all possible objects correctly", () => {
    const samples: Statuses[] = [
      ["user", "D", "private", "ongoing", "event", "active", "punctual"],
      ["admin", "S", "public", "done", "class", "archive", "overdue"],
    ];

    const tree = samples.map((x) => arrayToObject(x));
    expect(tree).toMatchSnapshot();
  });
});
