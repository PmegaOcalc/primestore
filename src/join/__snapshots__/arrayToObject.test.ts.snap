// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrayToObject function renders all possible objects correctly 1`] = `
[
  {
    "D": true,
    "active": true,
    "event": true,
    "ongoing": true,
    "private": true,
    "punctual": true,
    "user": true,
  },
  {
    "S": true,
    "admin": true,
    "archive": true,
    "class": true,
    "done": true,
    "overdue": true,
    "public": true,
  },
]
`;
